### PR TITLE
fix(core): menu accessibility

### PR DESCRIPTION
**Pull request by dimamarksman**
## Related Issue(s)

closes https://github.com/sap/fundamental-ngx/issues/6210

## Description

This Pr fixes some accessibility issues related to `fd-menu` component

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [na] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [na] update `README.md`
-   [na] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/menu/menu.component.ts [hunk 1]
```diff
@@ -135,6 +157,23 @@ export class MenuComponent
...
+         * so menu.isOpen will be up to date.
+         *
+         * This is matter cause popoverService.initialise() copies
+         * "isOpenChange" event emitter and fires this
+         * before menu.isOpen gets actually updated.
+         *
+         * TODO: Needs refactoring to make things obvious
+         * and avoid such copying.
+         */
+        this.isOpenChange.subscribe((isOpen) => {
```
Comment (Betrozov) [inline libs/core/src/lib/menu/menu.component.ts:173]: do we need unsubscribe?

File: libs/core/src/lib/menu/menu.component.ts [hunk 2]
```diff
@@ -162,14 +195,14 @@ export class MenuComponent
 
     /** @hidden */
     ngAfterViewInit(): void {
+        this._menuService.setMenuMode(this.mobile);
         this._menuService.isMobileMode.subscribe((isMobile) => {
```
Comment (Betrozov) [inline libs/core/src/lib/menu/menu.component.ts:199]: same

File: libs/core/src/lib/popover/popover.component.ts [hunk 3]
```diff
@@ -281,8 +281,6 @@ export class PopoverComponent
             injector
         );
 
-        console.log(111111, this._mobileModeComponentRef);
```
Comment (Betrozov) [inline libs/core/src/lib/popover/popover.component.ts:284]: 😅

**Review by InnaAtanasova — APPROVED** [review]

File: libs/core/src/lib/popover/popover.component.ts (hunk 3 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/popover/popover.component.ts:284]: +1

**Comment by InnaAtanasova** [comment]
Please address Yury's comments, but the rest looks good so I approved.

File: libs/core/src/lib/menu/menu.component.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/core/src/lib/menu/menu.component.ts]: added, thanks

File: libs/core/src/lib/menu/menu.component.ts (hunk 2 above)
Comment (dimamarksman) [reply libs/core/src/lib/menu/menu.component.ts:201]: added isMobileMode.complete()

**Review by Betrozov — APPROVED** [review]

**Comment by Sanjay-1312** [comment]
Issue 1: Fixed.

Issue 2: Not fixed.
image
Observed behavior: The list and list count is not read for the sub menu of the list.
Expected Behavior: The sub menu items and their list count should be read to the user.

Issue 3: Fixed.

Issue 4: FIxed,

Issue 5: Fixed.

Issue 6: Fixed.

issue 7: Fixed.

Retest Details:
https://deploy-preview-7382--fundamental-ngx.netlify.app/#/core/menu
JAWS 2021

**Review by platon-rov — APPROVED** [review]

**Comment by dimamarksman** [comment]
> Issue 2:
> image
> Observed behavior: The list and list count is not read for the sub menu of the list.
> Expected Behavior: The sub menu items and their list count should be read to the user.

@Sanjay-1312 this should be fixed, please check one more time
I see this:
image

**Comment by dimamarksman** [comment]
reopen cause waiting for QA team

**Comment by Veerakarthick** [comment]
Hello Colleague,

Retest has been taken below are the details.

Issue1: Fixed.
Issue2: Fixed.
Issue3: fixed
issue4: fixed
Issue5: fixed
issue6:Not reproducible.
Issue7:
Observed Behavior: Menu button names are truncated when text spacing is enabled.
Expected Behavior: Ideally, UI names should not be truncated when text spacing is applied. Screenshot:

Regards,
Karthikeyan.c

**Pull request closed without merging**